### PR TITLE
Refactor apollo integration

### PR DESCRIPTION
**Pull request by znarf**
As we're not gonna merge https://github.com/opencollective/opencollective-frontend/pull/5024, extract some interesting refactoring out of it.

### CODE REVIEW

**Comment by znarf** [comment]
Could that break `FileDownloader`? (`this.props.url`)

**Comment by znarf** [comment]
Should we delete the `dataIdFromObject` thing?

File: lib/apollo-client.js [hunk 1]
```diff
@@ -6,22 +6,50 @@ import { setContext } from '@apollo/client/link/context';
...
+  const apiKey = !process.browser ? process.env.API_KEY : null;
+  return `${getBaseApiUrl()}/graphql${apiVersion ? `/${apiVersion}` : ''}${apiKey ? `?api_key=${apiKey}` : ''}`;
+};
+
+const getCustomAgent = () => {
+  if (!customAgent) {
+    const http = require('http');
+    const https = require('https');
+    const httpAgent = new http.Agent({ keepAlive: true });
+    const httpsAgent = new https.Agent({ keepAlive: true });
```
Comment (Betree) [inline lib/apollo-client.js:35]: Would be nice to have an expiration on the keepAlive (2min? 5min?) to handle certificate rotation. Unless it's done automatically?

**Review by Betree — APPROVED** [review]
:tada:

**Pull request merged**